engine/src/test/java/com/datatorrent/stram/util/SecurityUtilsTest.java

```diff
@@ -34,12 +35,30 @@
   public void testStramWebSecurity()
   {
     checkWebSecurity(false, false);
-    Configuration conf = new Configuration();
+    Configuration conf = setupConfiguration(null);
     checkSecurityConfiguration(conf, new boolean[][]{{false, false}, {false, true}, {false, false}, {false, false}, {false, false}});
-    conf.set(SecurityUtils.HADOOP_HTTP_AUTH_PROP, "kerberos");
+    conf = setupConfiguration(AuthScheme.SPNEGO);
     checkSecurityConfiguration(conf, new boolean[][]{{true, false}, {true, true}, {true, false}, {true, false}, {true, true}});
   }
 
+  @Test
+  public void testInitAuth() throws NoSuchFieldException, IllegalAccessException
+  {
+    Configuration conf = setupConfiguration(AuthScheme.BASIC);
+    SecurityUtils.init(conf);
+    WebServicesClientTest.checkUserCredentials("testuser", "testpass");
+  }
+
+  private Configuration setupConfiguration(AuthScheme authScheme)
+  {
+    Configuration conf = new Configuration();
+    if (authScheme != null) {
+      conf.set(SecurityUtils.HADOOP_HTTP_AUTH_PROP, authScheme.getName());
+      conf.addResource("security/dt-site-" + authScheme.getName() + ".xml");
```

Comment:
Since we have been using dt-site.xml everywhere including the engine code, thought it would be better to leave it consistent. We can change it if we make an overall change to apex-site.